src/win/getaddrinfo.c

```diff
@@ -103,17 +102,7 @@ static void uv__getaddrinfo_work(struct uv__work* w) {
  * Each size calculation is adjusted to avoid unaligned pointers.
  */
 static void uv__getaddrinfo_done(struct uv__work* w, int status) {
-  uv_getaddrinfo_t* req;
-  size_t addrinfo_len = 0;
-  ssize_t name_len = 0;
-  size_t addrinfo_struct_len = ALIGNED_SIZE(sizeof(struct addrinfo));
-  struct addrinfoW* addrinfow_ptr;
-  struct addrinfo* addrinfo_ptr;
-  char* alloc_ptr = NULL;
-  char* cur_ptr = NULL;
-  int r;
-
-  req = container_of(w, uv_getaddrinfo_t, work_req);
+  uv_getaddrinfo_t *req = container_of(w, uv_getaddrinfo_t, work_req);
```

Comment:
General style nit (applies here and elsewhere): star leans left, i.e.:
```suggestion
  uv_getaddrinfo_t* req = container_of(w, uv_getaddrinfo_t, work_req);
```
Another style nit: libuv code generally declares variables at the top of the function, not inline.

Reply:
I see, though this is generally discouraged nowadays, as reducing scope aids in reasoning about the code and gives clearer context.

Though in this case the variable is declared at the top of the function. It's also initialized in the declaration, something which the existing code already does.

Reply:
None of the variables in this scope appear to be mixing code with initialization (constant initialization is distinct from code-initialization in the C standards)

Reply:
In the C standard there is the concept of constant expressions, for example defined in §6.6 of draft C99 standard.
So I take it the rule would be along the lines of only allowing initialization from constant expressions, the same as what is only allowed for non-automatic variables, for example.

Reply:
Though that function did not previously violate that rule, the very next function does, and so many others. I will try not change anything then.